### PR TITLE
Simplify compiler support code

### PR DESCRIPTION
**Pull request by sas**

### CODE REVIEW

File: Headers/DebugServer2/Utils/CompilerSupport.h [hunk 1]
```diff
@@ -12,26 +12,7 @@
...
-#if !defined(__has_builtin)
-#if defined(COMPILER_GCC)
-#define __has_builtin(BUILTIN) 1
-#else
-#define __has_builtin(BUILTIN) 0
-#endif
-#endif
-
-#if __has_attribute(format)
+#if defined(COMPILER_GCC) || defined(COMPILER_CLANG)
```
Comment (xiaobai) [inline Headers/DebugServer2/Utils/CompilerSupport.h]: `#endif` is where?

File: Headers/DebugServer2/Utils/CompilerSupport.h [hunk 2]
```diff
@@ -66,23 +41,19 @@
...
-#error "Compiler attribute __packed__ is required."
 #endif
 
 #if defined(COMPILER_MSVC)
 #define DS2_ATTRIBUTE_ALIGNED "DS2_ATTRIBUTE_ALIGNED not implemented on MSVC"
-#elif __has_attribute(__aligned__)
+#elif defined(COMPILER_GCC) || defined(COMPILER_CLANG)
 #define DS2_ATTRIBUTE_ALIGNED(SIZE) __attribute__((__aligned__(SIZE)))
-#else
-#error "Compiler attribute __aligned__ is required."
```
Comment (xiaobai) [inline Headers/DebugServer2/Utils/CompilerSupport.h]: Might be worth keeping some of these `#errors` around, at least to tell the user that their compiler isn't supported.

File: Headers/DebugServer2/Utils/CompilerSupport.h (hunk 1 above)
Comment (xiaobai) [reply Headers/DebugServer2/Utils/CompilerSupport.h]: Wait, this becomes only for `__has_attribute(format)` now right?

File: Headers/DebugServer2/Utils/CompilerSupport.h (hunk 2 above)
Comment (sas) [reply Headers/DebugServer2/Utils/CompilerSupport.h]: If the compiler doesn't support it, you'll get errors during the build anyway.

File: Headers/DebugServer2/Utils/CompilerSupport.h (hunk 1 above)
Comment (sas) [reply Headers/DebugServer2/Utils/CompilerSupport.h]: I don't understand the question :p

Comment (xiaobai) [reply Headers/DebugServer2/Utils/CompilerSupport.h]: I misunderstood what this was replacing. It looks fine to me now.

File: Headers/DebugServer2/Utils/CompilerSupport.h (hunk 2 above)
Comment (xiaobai) [reply Headers/DebugServer2/Utils/CompilerSupport.h]: True.

**Comment by xiaobai** [comment]
Your test failure is related to watchpoints, which I bet is a flake. If you retry it, it'll probably pass.

**Pull request merged**